src/setup-beam.js

```diff
@@ -436,6 +436,8 @@ function getVersionFromSpec(spec0, versions0) {
       // If `version-type: strict` or version is RC, we obtain it directly
       version = versions0[spec]
     }
+  } else if (spec0 === 'latest') {
+    version = versions0[versions0.latest]
```

Comment:
This, and the change below, do not seem related to the title of the pull request (talks about adding, not fixing). Why is this required?

Reply:
If you take a look at the issue. Then you will see that currently there is a bug causing it to return 404, since we didn’t have any integration test, this was not caught.


Before
Erlang/OTP 27.0

After
Erlang/OTP OTP-27.0.1


The changes ensures that the proper version name is used and are tested.

Please notice the missing OTP-.

Reply:
> since we didn’t have any integration test

It should be possible to write a test that doesn't rely on online elements (similar to what you proposed with the fixed `builds.txt` elements). I'd rather prefer that than rely on remote inconsistencies outside of the action's control.

Reply:
There's this test:

```javascript
  spec = 'latest'
  expected = 'OTP-27.0'
  got = setupBeam.getVersionFromSpec(spec, matrix.otp['ubuntu-24.04'])
  assert.deepStrictEqual(got, expected)
```

Whatever it's not testing well, if there's missing elements, it should.

I don't see how a change to the code wouldn't change those tests results (that same function is being tested), but I didn't dig deep.

Reply:
These tests have no issues. I understand that you do not prefer integration tests, but an integration test would have caught an issue with getOTPVersions. Decoupling code and uniting testing is good practice, but with integration tests, we would know if there was a hidden contract between the layers.

As I said in my prior PR, I do not want to do any significant refactoring, regardless of how beneficial it might be. I'm just trying to make incremental changes to improve this codebase's health.